src/dnx.host/FrameworkNameUtility.cs

```diff
@@ -7,24 +7,44 @@
 
 namespace dnx.host
 {
+    /// <summary>
+    /// Parses framework names expected by the DNX _runtime_.
+    /// </summary>
+    /// <remarks>
+    /// Note: This does not need to be complete because there are a fixed set of frameworks
+    /// understood by the runtimes.
+    /// </remarks>
     internal class FrameworkNameUtility
     {
-        internal static FrameworkName ParseFrameworkName(string frameworkName)
+        internal static bool TryParseFrameworkName(string frameworkName, out FrameworkName parsed)
```

Comment:
Add support for all known .NET framework versions. .NET 4.5 .NET 4.5.1, .NET 4.5.2 .NET 4.6